### PR TITLE
added all tasks

### PR DESCRIPTION
**Pull request by mstefanchuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/AbstractFigure.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    private String color;
+
+    public AbstractFigure(String color) {
+        this.color = color;
+    }
+
+    @Override
+    public String getColor() {
+        return color;
+    }
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/AbstractFigure.java:13]: Its a getter for one common property you can not put that method into interface

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class ColorSupplier {
+    public String getRandomColor() {
+        String[] colors = {"blue", "red", "yellow", "green", "orange", "purple"};
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: better to create Enum for colors

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                return new IsoscelesTrapezoid(random.nextDouble() * 10
+                        + 1, random.nextDouble() * 10 + 1, random
+                        .nextDouble() * 10 + 1, colorSupplier.getRandomColor());
+            default:
+                return null;
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, "white");
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/FigureSupplier.java:37]: Make all magic numbers a Constants

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                return new RightTriangle(random.nextDouble() * 10
+                        + 1, random.nextDouble() * 10 + 1, colorSupplier.getRandomColor());
+            case 3:
+                return new Circle(random.nextDouble() * 10 + 1, colorSupplier.getRandomColor());
+            case 4:
+                return new IsoscelesTrapezoid(random.nextDouble() * 10
+                        + 1, random.nextDouble() * 10 + 1, random
+                        .nextDouble() * 10 + 1, colorSupplier.getRandomColor());
+            default:
+                return null;
```
Comment (Vasul-Pankiv) [inline src/main/java/core/basesyntax/FigureSupplier.java:32]: I think it better to return the default figure

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        this.colorSupplier = colorSupplier;
+    }
+
+    public AbstractFigure getRandomFigure() {
+        Random random = new Random();
+        int figureType = random.nextInt(FigureType.values().length);
+
+        switch (figureType) {
+            case 0:
+                return new Square(random.nextDouble() * 10 + 1, colorSupplier.getRandomColor());
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:20]: let's extract this reapeated logic `random.nextDouble() * 10 + 1` to a private method, and preferably 10 to a constant

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public AbstractFigure getRandomFigure() {
+        Random random = new Random();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:15]: make random a class field, same as color supplier

File: src/main/java/core/basesyntax/Constants.java [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package core.basesyntax;
+
+public class Constants {
+    public static final int DEFAULT_RADIUS = 10;
+    public static final String DEFAULT_COLOR = "white";
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Constants.java:5]: remove this class, move constants to figureSupplier
white - don't hardcode, use enum

File: src/main/java/core/basesyntax/FigureType.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public enum FigureType {
+    SQUARE, RECTANGLE, RIGHT_TRIANGLE, CIRCLE, ISOSCELES_TRAPEZOID
+}
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureType.java:5]: it's redundant, you don't use those values, just generated based off integers

File: src/main/java/core/basesyntax/Main.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+        AbstractFigure[] figures = new AbstractFigure[3];
+
+        for (int i = 0; i < 3; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (AbstractFigure figure : figures) {
+            figure.draw();
+        }
+    }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Main.java:19]: read the task description again
don't hardcode lengths 2+ times, use figures.length
you need only one for loop in this class
Half of figures should be random, the other half - default

File: src/main/java/core/basesyntax/service/Color.java [hunk 10]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.service;
+
+public enum Color {
+    RED, GREEN, BLUE, YELLOW, PURPLE, ORANGE;
+
+    public static Color getRandomColor() {
+        return values()[(int) (Math.random() * values().length)];
+    }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/service/Color.java:8]: this logic should be in color supplier, not here

**Review by okuzan — CHANGES_REQUESTED** [review]
add package for figures
color isn't/shouldn't be a service

File: src/main/java/core/basesyntax/Constants.java (hunk 7 above)
Comment (olekskov) [inline src/main/java/core/basesyntax/Constants.java:5]: totally agree with Oleh's comment
these constants are belong to FigureSupplier and should be there

File: src/main/java/core/basesyntax/FigureSupplier.java (hunk 6 above)
Comment (olekskov) [inline src/main/java/core/basesyntax/FigureSupplier.java:15]: again not fixed comment from previous review
random should be class variable in order not to create new object each time method is called

File: src/main/java/core/basesyntax/FigureSupplier.java (hunk 5 above)
Comment (olekskov) [inline src/main/java/core/basesyntax/FigureSupplier.java:20]: and again
random.nextDouble() * 10 + 1 expression is used so many times so it deserves to be in a separate method with proper naming

File: src/main/java/core/basesyntax/FigureType.java (hunk 8 above)
Comment (olekskov) [reply src/main/java/core/basesyntax/FigureType.java:5]: not fixed

**Review by olekskov — CHANGES_REQUESTED** [review]
Please fix all comments from previous review

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static class Constants {
+        public static final int DEFAULT_RADIUS = 10;
+        public static final String DEFAULT_COLOR = "white";
+    }
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:10]:
```suggestion
        private static final int DEFAULT_RADIUS = 10;
        private static final String DEFAULT_COLOR = Color.WHITE.name();
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    public AbstractFigure getRandomFigure() {
+        int figureType = random.nextInt(5);
+
+        switch (figureType) {
+            case 0:
+                return new Square(getRandomLength(), colorSupplier.getRandomColor());
+            case 1:
+                return new Rectangle(getRandomLength(), getRandomLength(), colorSupplier
+                        .getRandomColor());
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:27]:
```suggestion
                return new Rectangle(getRandomLength(), getRandomLength(), 
                               colorSupplier.getRandomColor());
```
a bit weird line breaking imo

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+                return new RightTriangle(getRandomLength(), getRandomLength(), colorSupplier
+                        .getRandomColor());
+            case 3:
+                return new Circle(getRandomLength(), colorSupplier.getRandomColor());
+            case 4:
+                return new IsoscelesTrapezoid(getRandomLength(), getRandomLength(),
+                        getRandomLength(), colorSupplier.getRandomColor());
+            default:
+                return getDefaultFigure();
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:38]:
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        public static final int DEFAULT_RADIUS = 10;
+        public static final String DEFAULT_COLOR = "white";
+    }
+
+    private ColorSupplier colorSupplier;
+    private Random random;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+        this.random = new Random();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:16]:
```suggestion
        random = new Random();
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public AbstractFigure getRandomFigure() {
+        int figureType = random.nextInt(5);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:20]:
```suggestion
        int figureType = random.nextInt(FIGURE_COUNT);
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+            case 2:
+                return new RightTriangle(getRandomLength(), getRandomLength(), colorSupplier
+                        .getRandomColor());
+            case 3:
+                return new Circle(getRandomLength(), colorSupplier.getRandomColor());
+            case 4:
+                return new IsoscelesTrapezoid(getRandomLength(), getRandomLength(),
+                        getRandomLength(), colorSupplier.getRandomColor());
+            default:
+                return getDefaultFigure();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:37]:
```suggestion
            default:
                return new IsoscelesTrapezoid(getRandomLength(), getRandomLength(),
                        getRandomLength(), colorSupplier.getRandomColor());
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+
+        }
+    }
+
+    private double getRandomLength() {
+        return random.nextDouble() * 10 + 1;
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:43]:
```suggestion
        return random.nextDouble() * MAX_LENGTH + 1;
```

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java (hunk 15 above)
Comment (mstefanchuk) [reply src/main/java/core/basesyntax/FigureSupplier.java:20]: Тобто тепер потрібно оголосити цю змінну в цьому класі, правильно я розумію?

**Review by okuzan — APPROVED** [review]
Well done

**Pull request closed without merging**